fab/fabfile.py

```diff
@@ -107,66 +109,49 @@ def _require_target():
 
 
 class DeployMetadata(object):
-    def __init__(self, env):
-        self.env = env
+    def __init__(self):
```

Comment:
There's kind of a weird interdependency thing going on between `DeployMetadata` and `env`, where they each depend on the other.  Here `DeployMetadata` accesses only `code_root` and `environment` from the `env` object, but it accesses it as a global.  It would be a relatively small matter to require those two as simple arguments on initialization, so the object would be self-contained.